### PR TITLE
Feature/219 accs delete setting

### LINKED ISSUE #219: Use setting to determine which type of accounts to delete if they have no children
Currently we delete them if they are Household or Administrative account and all children have been deleted, but this should be configurable.

### LINKED ISSUE #220: Use setting to determine which type of accounts support seasonal addresses
For this and similar functionality we should probably have a method that gets an account record type ID and returns true or false depending on whether functionality is active for that type. Something like: 

public static Boolean isSeasonalAddrActiveForRecType(ID recTypeID)

### PR DESCRIPTION
**Pull request by ceiroa**
# Warning

# Info
- Multi-addresses now supported in any Account record type existing in the org.
- Admin can how choose with type of Accounts will be automatically deleted if all its children Contacts have been deleted.

# Issues
Fixes #219
Fixes #220

### CODE REVIEW

**Comment by ceiroa** [comment]
Hi @kbromer ~ There are a lot of changes that went into this pull request, including a new reusable lightning component, and some Javascript refactoring. Do you think you could review it for me? Thanks!

File: src/aura/STG_CMP_Base/STG_CMP_BaseHelper.js [hunk 1]
```diff
@@ -81,6 +56,14 @@
 		}
 	},
 	
+	addPrefix : function(namespacePrefix, key) {
```
Comment (davidhabib) [inline src/aura/STG_CMP_Base/STG_CMP_BaseHelper.js]: I assume the namespace prefix contains the dot in all this code?

File: src/aura/STG_CMP_System/STG_CMP_System.cmp [hunk 2]
```diff
@@ -14,10 +13,10 @@
             <aura:if isTrue="{!v.isView}">
                 <ui:outputText aura:id="recTypeName" value="{!v.accRecTypeName}"/>
             <aura:set attribute="else">
-                <ui:inputSelect aura:id="accRecTypes">
+                <ui:inputSelect aura:id="accRecTypes" change="{!c.onSelectChange}">
                     <aura:iteration items="{!v.accRecTypes}" var="recType">
-                        <ui:inputSelectOption label="{!recType.devName}" text="{!recType.id}" 
-                            value="{!v.accRecTypeName == recType.devName ? true : false}" />
+                        <ui:inputSelectOption label="{!recType.name}" text="{!recType.id}" 
+                            value="{!v.accRecTypeId == recType.id ? true : false}" />
```
Comment (davidhabib) [inline src/aura/STG_CMP_System/STG_CMP_System.cmp]: why the change from using recordtype Developer names to using recordtype Id's?

File: src/package.xml [hunk 3]
```diff
@@ -214,15 +215,15 @@
...
         <members>Hierarchy_Settings__c.Account_Processor__c</members>
+        <members>Hierarchy_Settings__c.Accounts_to_Delete__c</members>
         <members>Hierarchy_Settings__c.Affl_ProgEnroll_Del_Status__c</members>
         <members>Hierarchy_Settings__c.Affl_ProgEnroll_Del__c</members>
         <members>Hierarchy_Settings__c.Allow_AutoCreated_Duplicates__c</members>
         <members>Hierarchy_Settings__c.Disable_Error_Handling__c</members>
         <members>Hierarchy_Settings__c.Enable_Debug__c</members>
         <members>Hierarchy_Settings__c.Error_Notifications_On__c</members>
         <members>Hierarchy_Settings__c.Error_Notifications_To__c</members>
-        <members>Hierarchy_Settings__c.Household_Adm_Acct_Addresses_Enabled__c</members>
```
Comment (davidhabib) [inline src/package.xml]: can you delete these if they were already packaged?

File: src/aura/STG_CMP_Base/STG_CMP_BaseHelper.js (hunk 1 above)
Comment (ceiroa) [inline src/aura/STG_CMP_Base/STG_CMP_BaseHelper.js]: Actually, we are manipulating custom settings fields, and the namespace needs to have the double underscore. This is done in https://github.com/SalesforceFoundation/HEDAP/blob/master/src/pages/STG_Settings.page#L29.

File: src/aura/STG_CMP_System/STG_CMP_System.cmp (hunk 2 above)
Comment (ceiroa) [inline src/aura/STG_CMP_System/STG_CMP_System.cmp]: Just to standardize storing recordtype IDs in all settings. The change to using dev names was done recently (https://github.com/SalesforceFoundation/HEDAP/pull/226/), as an intermediate step.

File: src/package.xml (hunk 3 above)
Comment (ceiroa) [inline src/package.xml]: I think you can delete custom settings fields the same you can delete object fields.

**Comment by ceiroa** [comment]
@davidhabib ~ This is ready for another look. I moved those checkboxes we talked about to the right, and since I was on it, I implemented the same pattern for all checkboxes across settings. The others didn't have labels, but this make them all have the slds look also in edit mode.
